include/logging/log_backend.h

```diff
@@ -95,6 +96,45 @@ extern const struct log_backend __log_backends_end[];
 	}
 
 
+/**
+ * @brief Initialize or initiate the logging backend.
+ *
+ * If backend initialization takes longer time it could block logging thread
+ * if backend is autostarted. That is because all backends are initilized in
+ * the context of the logging thread. In that case, backend shall provide
+ * function for polling for readiness (@ref log_backend_is_ready).
+ *
+ * @param[in] backend  Pointer to the backend instance.
+ */
+static inline void log_backend_init(const struct log_backend *const backend)
+{
+	__ASSERT_NO_MSG(backend != NULL);
+	if (backend->api->init) {
+		backend->api->init(backend);
+	}
+}
+
+/**
+ * @brief Poll for backend readiness.
+ *
+ * If backend is ready immediately after initialization then backend may not
+ * provide this function.
+ *
+ * @param[in] backend  Pointer to the backend instance.
+ *
+ * @retval 0 if backend is ready.
+ * @retval -EBUSY if backend is not yet ready.
+ */
+static inline bool log_backend_is_ready(const struct log_backend *const backend)
+{
+	__ASSERT_NO_MSG(backend != NULL);
+	if (backend->api->is_ready) {
+		return backend->api->is_ready(backend);
+	}
+
+	return 0;
+}
```

Comment:
Since this is boolean, shouldn't returning `true` when backend is ready be conceptually easier to understand?

Reply:
from the description of the function it doesn't sound like it should be bool
```
 * @param[in] backend  Pointer to the backend instance.
 *
 * @retval 0 if backend is ready.
 * @retval -EBUSY if backend is not yet ready.
```